android/app/src/main/java/com/microsoft/reactnativecodepush/CodePush.java

```diff
@@ -0,0 +1,412 @@
+        this.codePushConfig = new CodePushConfig(deploymentKey, this.applicationContext);
+        checkForPendingUpdate(/*needsRestart*/ false);
+    }
+
+    public ReactPackage getReactPackage() {
+        if (codePushReactPackage == null) {
+            codePushReactPackage = new CodePushReactPackage();
+        }
+        return codePushReactPackage;
+    }
+
+    public String getDocumentsDirectory() {
+        return codePushPackage.getDocumentsDirectory();
+    }
+
+    public String getBundleUrl(String assetsBundleFileName) {
+        this.assetsBundleFileName = assetsBundleFileName;
+        String binaryJsBundleUrl = ASSETS_BUNDLE_PREFIX + assetsBundleFileName;
+        try {
+            String packageFile = codePushPackage.getCurrentPackageBundlePath();
+            if (packageFile == null) {
+                // There has not been any downloaded updates.
+                return binaryJsBundleUrl;
+            }
+
+            return packageFile;
+        } catch (IOException e) {
+            throw new CodePushUnknownException("Error in getting current package bundle path", e);
+        }
+    }
+
+    private void cancelRollbackTimer() {
+        if(timer != null) {
+            timer.cancel();
+            timer = null;
+        }
+    }
+
+    private void checkForPendingUpdate(boolean needsRestart) {
+        SharedPreferences settings = applicationContext.getSharedPreferences(CODE_PUSH_PREFERENCES, 0);
+        String pendingUpdateString = settings.getString(PENDING_UPDATE_KEY, null);
+
+        if (pendingUpdateString != null) {
+            try {
+                JSONObject pendingUpdateJSON = new JSONObject(pendingUpdateString);
+                String pendingHash = pendingUpdateJSON.getString(PENDING_UPDATE_HASH_KEY);
+                String currentHash = codePushPackage.getCurrentPackageHash();
+                if (pendingHash.equals(currentHash)) {
+                    int rollbackTimeout = pendingUpdateJSON.getInt(PENDING_UPDATE_ROLLBACK_TIMEOUT_KEY);
+                    initializeUpdateWithRollbackTimeout(rollbackTimeout, needsRestart);
+
+                    settings.edit().remove(PENDING_UPDATE_KEY).commit();
+                }
+            } catch (JSONException e) {
+                // Should not happen.
+                throw new CodePushUnknownException("Unable to parse pending update metadata " +
+                        pendingUpdateString + " stored in SharedPreferences", e);
+            } catch (IOException e) {
+                // There is no current package hash.
+                throw new CodePushUnknownException("Should not register a pending update without a saving a current package", e);
+            }
+        }
+    }
+
+    private void checkForPendingUpdateDuringResume() {
+        if (resumablePendingUpdateAvailable) {
+            checkForPendingUpdate(/*needsRestart*/ true);
+        }
+    }
+
+    private WritableMap constantsToExport() {
```

Comment:
What is this method for? Doesn't the override in the native module below handle this?